### ISSUE TITLE
Improve consistency of the Cancel/OK button pairs

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#50518]
### Description

In various parts of the UI, Cancel/OK button pairs are in use where the 'OK' button is the primary action (Save, Apply, Create, etc.). These button paris are mostly used within modal dialogs and popups. Turns out there's some inconsistency especially about the 'Cancel' button type / variant: sometimes it's a secondary button, sometimes a tertiary one. There's no apparent reason for this difference. Also, sometimes the alignment and spacing is not consistent.

- For better consistency, the 'Cancel' buttons should always be either secondary or tertiary. 
- The alignment / spacing should be standardized.

Maybe consider to create a new UI component to ensure consistency and avoid repetitive code.

A few example screenshots:

Secondary:

<img width="527" alt="secondary create template part" src="https://github.com/WordPress/gutenberg/assets/1682452/739ea590-41cd-4e53-8738-3f9a5da0173f">

Tertiary: 

<img width="388" alt="tertiary create reusable block" src="https://github.com/WordPress/gutenberg/assets/1682452/dd8cbda7-23e3-46e2-a086-3a18f08991a1">

Tertiary: 

<img width="489" alt="tertiary lock block" src="https://github.com/WordPress/gutenberg/assets/1682452/3ccb0e23-302a-49fb-a307-7a8ac0862e1b">

Tertiary: 

<img width="504" alt="tertiary external image" src="https://github.com/WordPress/gutenberg/assets/1682452/fba190d6-640d-4462-b31e-4b77b975e6bc">

No type specified and inconsistent alignment. This will be likely fixed in https://github.com/WordPress/gutenberg/pull/50384

<img width="634" alt="no class classic block" src="https://github.com/WordPress/gutenberg/assets/1682452/9843baab-18b3-4ec4-af22-07fca7364875">

Tertiary: 

<img width="396" alt="tertiary navigation ad link popup" src="https://github.com/WordPress/gutenberg/assets/1682452/2b86e76a-62d2-47a0-97b8-305dd6646640">

Secondary:

<img width="413" alt="secondary delete menu" src="https://github.com/WordPress/gutenberg/assets/1682452/ad43f84c-1dc5-40a2-ad0d-e01bf0189e3a">

Tertiary: 

<img width="502" alt="tertiary convert to inks" src="https://github.com/WordPress/gutenberg/assets/1682452/827ea6b9-276f-4bc4-853e-b1a6d707a3b8">

Tertiary: 

<img width="409" alt="tertiary create custom template" src="https://github.com/WordPress/gutenberg/assets/1682452/ee58644a-5041-4b6e-bd08-223030182967">

Tertiary: 

<img width="411" alt="tertiary rename tp" src="https://github.com/WordPress/gutenberg/assets/1682452/b0ff6e74-d573-499c-8a14-abc9beb65b91">

Tertiary: 

<img width="730" alt="tertiary confirm" src="https://github.com/WordPress/gutenberg/assets/1682452/f2f72126-c17f-4cb3-8e4f-d97f09891878">

## Edge cases:

Edit site > VIew mode > Save modal dialog:

<img width="620" alt="edit site view mode save" src="https://github.com/WordPress/gutenberg/assets/1682452/a9f01d8f-ebdf-4e2c-8306-ec0dd3eb64d9">

Guide modal dialog: Previous, Next, and Finish buttons:

<img width="672" alt="Screenshot 2023-05-10 at 14 20 27" src="https://github.com/WordPress/gutenberg/assets/1682452/1f7e55d7-4213-4028-80c9-094edc5dbcaf">

## Note about the tertiary button accessibility:

Worth reminding the tertiary button isn't fully accessible as it relies only on color. In absence of color, this button is not distinguishable from normal text as there's no shape to identify the button or any other styling to do that (for example an underline).

### Step-by-step reproduction instructions

- Go through the UI and check the styling of the buttons in the screenshots above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Note: as I'm noticing more and more small inconsistencies in the UI, I'd like to propose to create a new dedicated GitHub issue label for that. Suggestions welcome.

**Comment by Mamaduka** [comment]
cc @WordPress/gutenberg-design

**Comment by paaljoachim** [comment]
I believe @richtabor would like to take a closer look at this issue.

**Comment by richtabor** [comment]
>  I'd like to propose to create a new dedicated GitHub issue label for that. 

I don't personally think we need an issue for inconsistencies, but certainly tagging @wordpress/gutenberg-design will help keep them surfaced.

**Comment by richtabor** [comment]
Very nice. 

#### Buttons

Bringing more consistency is what buttons are used for cancel/ok pairs would be the first step. Looks like some are using secondary, while other tertiary button styles. 

#### Labels 

I don't think the labels themselves need to be 1:1 across the board. 

They should _probably_ reference the action intended, as these are confirmations. I.e. "Delete Menu" confirmation modal of "Delete Menu", or simply "Delete", not "Confirm". Let's get clear on this before pushing PRs.

The cancel action should always be "Cancel", which looks the be the case (just varying button styles).

**Comment by joedolson** [comment]
For accessibility, I'd strongly prefer that these actions always used the secondary style. It's a better user experience to be able to recognize controls and know their control boundaries.

**Comment by afercia** [comment]
> For accessibility, I'd strongly prefer that these actions always used the secondary style. It's a better user experience to be able to recognize controls and know their control boundaries.

For reference: the lack of accessibility of the tertiary buttons is a long standing issue and it was reported long time ago in https://github.com/WordPress/gutenberg/issues/23890

**Comment by aduth** [comment]
A couple thoughts about the inconsistency:

- As noted in the original issue description, ideally we'd be using common UI components that handle these types of patterns consistently. For confirmation dialogs, this would be [ConfirmDialog](https://wordpress.github.io/gutenberg/?path=/docs/components-confirmdialog--docs) / [AlertDialog](https://wordpress.github.io/gutenberg/?path=/docs/design-system-components-alertdialog--docs). Handling this centrally in a single component also allows us to more easily make a widespread change like which button variants to pair.
- As far as I can tell, we don't have any documented guidance about button pairing relationships as a pattern. This absence of clear direction likely contributes to inconsistencies.

On accessibility:

- Since the issue was opened, button styling has been updated to apply an increased font weight in #70787, so it's not strictly accurate anymore that tertiary buttons rely on color alone. Granted, this effect is subtle and "boldness" isn't especially distinguishable as representing an interactive button.
- Does context help here, in how the buttons are paired together in a dedicated space at the bottom of an action dialog?

It seems like a discrete action item could be to document the pairing guidance, including any relevant considerations for context.

**Comment by joedolson** [comment]
Having some documentation to specify the preferred pattern would definitely help matters. I don't think anybody would argue that having every pattern use a different layout and order makes the whole interface less predictable.

Re: tertiary button design. Technically, they were set to `font-weight: 499`, and may or may not render as bold. Since this is unreliable, I would not consider it accurate to say that this part of the issue is resolved. 

Context helps; but because the context is highly variable - sometimes at the top, sometimes at the bottom, sometimes together, sometimes separated, the context itself is not something you can necessarily use as a reliable framework.
